source/jormungandr/jormungandr/parking_space_availability/car/common_car_park_provider.py

```diff
@@ -0,0 +1,116 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from __future__ import absolute_import, print_function, unicode_literals, division
+
+import logging
+import pybreaker
+import requests as requests
+
+from jormungandr import cache, app, utils, new_relic
+from jormungandr.parking_space_availability import AbstractParkingPlacesProvider
+from abc import abstractmethod
+
+
+class CommonCarParkProvider(AbstractParkingPlacesProvider):
+
+    def __init__(self, url, operators, dataset, timeout, **kwargs):
+
+        self.ws_service_template = url + '?dataset={}'
+        self.operators = [o.lower() for o in operators]
+        self.timeout = timeout
+        self.dataset = dataset
+        self.fail_max = kwargs.get('circuit_breaker_max_fail', app.config['CIRCUIT_BREAKER_MAX_CAR_PARK_FAIL'])
+        self.reset_timeout = kwargs.get('circuit_breaker_reset_timeout', app.config['CIRCUIT_BREAKER_CAR_PARK_TIMEOUT_S'])
+        self.breaker = pybreaker.CircuitBreaker(fail_max=self.fail_max, reset_timeout=self.reset_timeout)
+        self.log = logging.LoggerAdapter(logging.getLogger(__name__), extra={'dataset': self.dataset})
+
+        if kwargs.get('api_key'):
+            self.api_key = kwargs.get('api_key')
+
+    @abstractmethod
+    def process_data(self, data, poi):
+        pass
+
+    def get_informations(self, poi):
+        if not poi.get('properties', {}).get('ref'):
+            return
+
+        data = self._call_webservice(self.ws_service_template.format(self.dataset))
+
+        if data:
+            return self.process_data(data, poi)
+
+    def support_poi(self, poi):
+        properties = poi.get('properties', {})
+        return properties.get('operator', '').lower() in self.operators
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_CAR_PARK', 30))
+    def _call_webservice(self, request_url):
+        try:
+            if self.api_key:
+                headers = {'Authorization': 'apiKey {}'.format(self.api_key)}
+            else:
+                headers = None
+            data = self.breaker.call(requests.get, url=request_url, headers=headers, timeout=self.timeout)
+            # record in newrelic
+            self.record_call("OK")
+            return data.json()
+        except pybreaker.CircuitBreakerError as e:
+            msg = '{} service dead (error: {})'.format(self.provider_name, e)
```

Comment:
After investigation, the message handling and new relic event recording can be factorized a bit. I suugest that this can be handled in a technical debt tocket because it goes beyond this PR.